src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -601,6 +601,7 @@ void Instance::OnFailSafeTimerExpired()
 
     ChipLogDetail(Zcl, "Failsafe timeout, tell platform driver to revert network credentials.");
     mpWirelessDriver->RevertConfiguration();
+    mAsyncCommandHandle.Release();
```

Comment:
Oh, and can we add some tests for this?  We should be able to add YAML tests for two cases:

1) Start a fail-safe timer, do some network manipulation, force-expire the failsafe by resetting it to 0, check that our network manipulation was reverted.
2) Start a fail-safe timer, do some network manipulation, issue a reboot command in yaml, wait for the device, check that our network manipulation was reverted.

Reply:
This PR is mostly about OpenThread which is used on embedded platforms only for which we don't have any CI tests. I can create another PR to verify if the failsafe works on Linux with Thread enabled and add Cirque tests for that.